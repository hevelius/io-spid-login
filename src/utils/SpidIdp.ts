interface SpidIdp {
  id: string;
  name: string;
  logo: string;
  profileUrl: string;
}

export type {SpidIdp};
