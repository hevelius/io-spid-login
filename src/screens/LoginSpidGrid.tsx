import React, {Component} from 'react';
import {View, StyleSheet} from 'react-native';
import {SpidIdp} from '../utils/SpidIdp';
import IdpsGrid from '../components/IdpsGrid';

interface Props {
  idps: Array<SpidIdp>;
  cols?: number;
  template?: (_: SpidIdp) => React.ReactElement;
  idpSelected?: (_: string) => void;
}

const styles = StyleSheet.create({
  gridContainer: {
    padding: 20,
    alignContent: 'center',
  },
});

export class LoginSpidGrid extends Component<Props> {
  constructor(props: Props) {
    super(props);
  }

  private onIdpSelected = (idp: SpidIdp): void => {
    this.props.idpSelected ? this.props.idpSelected(idp.id) : undefined;
  };

  render() {
    return (
      <View style={styles.gridContainer}>
        <IdpsGrid
          cols={this.props.cols}
          idps={[...this.props.idps]}
          onIdpSelected={this.onIdpSelected}
          template={this.props.template}
        />
      </View>
    );
  }
}
