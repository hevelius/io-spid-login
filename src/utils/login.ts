import { SessionToken } from "../utils/SessionToken";
import URLParse from "url-parse";
import { WebViewNavigation } from 'react-native-webview';

type LoginSuccess = {
    success: true;
    token: SessionToken;
};
  
type LoginFailure = {
    success: false;
    errorCode?: string;
};

const LOGIN_SUCCESS_PREFIX = "/profile.html";
const LOGIN_FAILURE_PREFIX = "/error.html";
const LOGIN_FAILURE_WITH_ERROR_CODE_PREFIX = "/error.html?errorCode=";

type LoginResult = LoginSuccess | LoginFailure;

  export const extractLoginResult = (url: string): LoginResult | undefined => {
    const test = URLParse(url, true);
    if (test.pathname === LOGIN_SUCCESS_PREFIX && test.query.token) {
      const token = test.query.token;
      if (token && token.length > 0) {
        return { success: true, token: token as SessionToken };
      } else {
        return { success: false };
      }
    }
  â€‹
    // Check for LOGIN_FAILURE
    if (url.indexOf(LOGIN_FAILURE_PREFIX) !== -1) {
      const failureWithErrorCodeTokenPathPos = url.indexOf(
        LOGIN_FAILURE_WITH_ERROR_CODE_PREFIX
      );
      // try to extract error code
      if (failureWithErrorCodeTokenPathPos !== -1) {
        const errCode = url.substr(
          failureWithErrorCodeTokenPathPos +
            LOGIN_FAILURE_WITH_ERROR_CODE_PREFIX.length
        );
        return {
          success: false,
          errorCode: errCode.length > 0 ? errCode : undefined
        };
      }
      return {
        success: false,
        errorCode: undefined
      };
    }
    // Url is not LOGIN related
    return undefined;
  };

  /**
 * Extract the login result from the given url.
 * Return true if the url contains login pattern & token
 */
export const onLoginUriChanged = (
    onFailure: (errorCode: string | undefined) => void,
    onSuccess: (_: SessionToken) => void
  ) => (navState: WebViewNavigation): boolean => {
    if (navState.url) {
      // If the url is not related to login this will be `null`
      const loginResult = extractLoginResult(navState.url);
      if (loginResult) {
        if (loginResult.success) {
          // In case of successful login
          onSuccess(loginResult.token);
          return true;
        } else {
          // In case of login failure
          onFailure(loginResult.errorCode);
        }
      }
    }
    return false;
  };