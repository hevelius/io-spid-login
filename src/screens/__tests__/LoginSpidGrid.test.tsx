import React from 'react';
import {View, Text} from 'react-native';
import renderer from 'react-test-renderer';

import {LoginSpidGrid} from '../LoginSpidGrid';
import {SpidIdp} from '../../utils/SpidIdp';

const generateIdps = (): Array<SpidIdp> => {
  return [
    {
      id: 'arubaid',
      name: 'Aruba',
      logo: 'https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-arubaid.png',
      profileUrl: 'http://selfcarespid.aruba.it',
    },
    {
      id: 'infocertid',
      name: 'Infocert',
      logo: 'https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-infocertid.png',
      profileUrl: 'https://my.infocert.it/selfcare',
    },
    {
      id: 'intesaid',
      name: 'Intesa',
      logo: 'https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-intesaid.png',
      profileUrl: 'https://spid.intesa.it',
    },
    {
      id: 'lepidaid',
      name: 'Lepida',
      logo: 'https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-lepidaid.png',
      profileUrl: 'https://id.lepida.it/',
    },
    {
      id: 'namirialid',
      name: 'Namirial',
      logo: 'https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-namirialid.png',
      profileUrl: 'https://idp.namirialtsp.com/idp',
    },
    {
      id: 'posteid',
      name: 'Poste',
      logo: 'https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-posteid.png',
      profileUrl: 'https://posteid.poste.it/private/cruscotto.shtml',
    },
    {
      id: 'sielteid',
      name: 'Sielte',
      logo: 'https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-sielteid.png',
      profileUrl: 'https://myid.sieltecloud.it/profile/',
    },
    {
      id: 'spiditalia',
      name: 'SPIDItalia Register.it',
      logo: 'https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-spiditalia.png',
      profileUrl: 'https://spid.register.it',
    },
    {
      id: 'timid',
      name: 'Telecom Italia',
      logo: 'https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-timid.png',
      profileUrl: 'https://id.tim.it/identity/private/',
    },
  ];
};

it('renders correctly with defaults', () => {
  const idpgrid = renderer
    .create(<LoginSpidGrid idps={generateIdps()} cols={2} />)
    .toJSON();
  expect(idpgrid).toMatchSnapshot();
});

it('renders correctly with custom template', () => {
  const templateGenerator = (item: SpidIdp): React.ReactElement => {
    return (
      <View style={{padding: 10, backgroundColor: 'red'}}>
        <Text>{item.name}</Text>
      </View>
    );
  };
  const idpgridtemplate = renderer
    .create(
      <LoginSpidGrid
        idps={generateIdps()}
        cols={2}
        template={templateGenerator}
      />,
    )
    .toJSON();
  expect(idpgridtemplate).toMatchSnapshot();
});
