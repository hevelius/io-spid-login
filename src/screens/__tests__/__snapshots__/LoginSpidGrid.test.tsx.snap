// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with custom template 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "padding": 20,
    }
  }
>
  <RCTScrollView
    bounces={false}
    data={
      Array [
        Object {
          "id": "arubaid",
          "logo": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-arubaid.png",
          "name": "Aruba",
          "profileUrl": "http://selfcarespid.aruba.it",
        },
        Object {
          "id": "infocertid",
          "logo": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-infocertid.png",
          "name": "Infocert",
          "profileUrl": "https://my.infocert.it/selfcare",
        },
        Object {
          "id": "intesaid",
          "logo": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-intesaid.png",
          "name": "Intesa",
          "profileUrl": "https://spid.intesa.it",
        },
        Object {
          "id": "lepidaid",
          "logo": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-lepidaid.png",
          "name": "Lepida",
          "profileUrl": "https://id.lepida.it/",
        },
        Object {
          "id": "namirialid",
          "logo": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-namirialid.png",
          "name": "Namirial",
          "profileUrl": "https://idp.namirialtsp.com/idp",
        },
        Object {
          "id": "posteid",
          "logo": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-posteid.png",
          "name": "Poste",
          "profileUrl": "https://posteid.poste.it/private/cruscotto.shtml",
        },
        Object {
          "id": "sielteid",
          "logo": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-sielteid.png",
          "name": "Sielte",
          "profileUrl": "https://myid.sieltecloud.it/profile/",
        },
        Object {
          "id": "spiditalia",
          "logo": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-spiditalia.png",
          "name": "SPIDItalia Register.it",
          "profileUrl": "https://spid.register.it",
        },
        Object {
          "id": "timid",
          "logo": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-timid.png",
          "name": "Telecom Italia",
          "profileUrl": "https://id.tim.it/identity/private/",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "padding": 10,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "red",
                    "padding": 10,
                  }
                }
              >
                <Text>
                  Aruba
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "padding": 10,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "red",
                    "padding": 10,
                  }
                }
              >
                <Text>
                  Infocert
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "padding": 10,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "red",
                    "padding": 10,
                  }
                }
              >
                <Text>
                  Intesa
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "padding": 10,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "red",
                    "padding": 10,
                  }
                }
              >
                <Text>
                  Lepida
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "padding": 10,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "red",
                    "padding": 10,
                  }
                }
              >
                <Text>
                  Namirial
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "padding": 10,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "red",
                    "padding": 10,
                  }
                }
              >
                <Text>
                  Poste
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "padding": 10,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "red",
                    "padding": 10,
                  }
                }
              >
                <Text>
                  Sielte
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "padding": 10,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "red",
                    "padding": 10,
                  }
                }
              >
                <Text>
                  SPIDItalia Register.it
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "padding": 10,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "red",
                    "padding": 10,
                  }
                }
              >
                <Text>
                  Telecom Italia
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "padding": 20,
    }
  }
>
  <RCTScrollView
    bounces={false}
    data={
      Array [
        Object {
          "id": "arubaid",
          "logo": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-arubaid.png",
          "name": "Aruba",
          "profileUrl": "http://selfcarespid.aruba.it",
        },
        Object {
          "id": "infocertid",
          "logo": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-infocertid.png",
          "name": "Infocert",
          "profileUrl": "https://my.infocert.it/selfcare",
        },
        Object {
          "id": "intesaid",
          "logo": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-intesaid.png",
          "name": "Intesa",
          "profileUrl": "https://spid.intesa.it",
        },
        Object {
          "id": "lepidaid",
          "logo": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-lepidaid.png",
          "name": "Lepida",
          "profileUrl": "https://id.lepida.it/",
        },
        Object {
          "id": "namirialid",
          "logo": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-namirialid.png",
          "name": "Namirial",
          "profileUrl": "https://idp.namirialtsp.com/idp",
        },
        Object {
          "id": "posteid",
          "logo": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-posteid.png",
          "name": "Poste",
          "profileUrl": "https://posteid.poste.it/private/cruscotto.shtml",
        },
        Object {
          "id": "sielteid",
          "logo": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-sielteid.png",
          "name": "Sielte",
          "profileUrl": "https://myid.sieltecloud.it/profile/",
        },
        Object {
          "id": "spiditalia",
          "logo": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-spiditalia.png",
          "name": "SPIDItalia Register.it",
          "profileUrl": "https://spid.register.it",
        },
        Object {
          "id": "timid",
          "logo": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-timid.png",
          "name": "Telecom Italia",
          "profileUrl": "https://id.tim.it/identity/private/",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "padding": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-arubaid.png",
                  }
                }
                style={
                  Object {
                    "height": 30,
                    "resizeMode": "contain",
                    "width": 120,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "padding": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-infocertid.png",
                  }
                }
                style={
                  Object {
                    "height": 30,
                    "resizeMode": "contain",
                    "width": 120,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "padding": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-intesaid.png",
                  }
                }
                style={
                  Object {
                    "height": 30,
                    "resizeMode": "contain",
                    "width": 120,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "padding": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-lepidaid.png",
                  }
                }
                style={
                  Object {
                    "height": 30,
                    "resizeMode": "contain",
                    "width": 120,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "padding": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-namirialid.png",
                  }
                }
                style={
                  Object {
                    "height": 30,
                    "resizeMode": "contain",
                    "width": 120,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "padding": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-posteid.png",
                  }
                }
                style={
                  Object {
                    "height": 30,
                    "resizeMode": "contain",
                    "width": 120,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "padding": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-sielteid.png",
                  }
                }
                style={
                  Object {
                    "height": 30,
                    "resizeMode": "contain",
                    "width": 120,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "padding": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-spiditalia.png",
                  }
                }
                style={
                  Object {
                    "height": 30,
                    "resizeMode": "contain",
                    "width": 120,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "padding": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/spid/idps/spid-idp-timid.png",
                  }
                }
                style={
                  Object {
                    "height": 30,
                    "resizeMode": "contain",
                    "width": 120,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
